devicetypes/smartthings/aeotec-doorbell-6.src/aeotec-doorbell-6.groovy

```diff
@@ -0,0 +1,225 @@
+	createEvent(name: "alarm", value: cmd.value ? "both" : "off")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	def endpoint = cmd.sourceEndPoint
+	if (endpoint == 1) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		String childDni = "${device.deviceNetworkId}:$endpoint"
+		def child = childDevices.find { it.deviceNetworkId == childDni }
+	}
+}
+
+def on() {
+	setActiveSound(1)
+	def cmds = []
+	cmds << zwave.basicV1.basicSet(value: 0xFF).format()
+	cmds
+}
+
+def off() {
+	setActiveSound(1)
+	def cmds = []
+	cmds << zwave.basicV1.basicSet(value: 0x00).format()
+	resetActiveSounds()
+	cmds
+}
+
+def strobe() {
+	on()
+}
+
+def siren() {
+	on()
+}
+
+def both() {
+	on()
+}
+
+def ping() {
+	zwave.basicV1.basicGet()
+}
+
+private addChildren(numberOfSounds) {
+	for (def endpoint : 2..numberOfSounds) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+			def componentLabel = device.displayName[0..-2] + "$endpoint"
+
+			addChildDevice("Aeotec Child Siren", childDni, device.getHub().getId(), [
+					completedSetup: true,
+					label         : componentLabel,
+					isComponent   : false,
+					componentName : "Aeotec Doorbell 6 $endpoint",
+					componentLabel: "Aeotec Doorbell 6 $endpoint"
+			])
+		} catch (Exception e) {
+			log.debug "Excep: ${e} "
+		}
+	}
+}
+
+private encap(cmd, endpoint = null) {
+	if (endpoint) {
+		zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+	} else {
+		cmd
+	}
+}
+
+def sendCommand(deviceDni, commands) {
+	def result = commands.collect {
+		if (it instanceof String) {
+			it
+		} else {
+			encap(it, channelNumber(deviceDni))
+		}
+	}
+	sendHubCommand(result, 100)
+}
+
+def channelNumber(String dni) {
+	dni?.split(":")[-1] as Integer
+}
+
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	if (cmd.notificationStatus == 0xFF) {
+		switch (cmd.event) {
+			case 0x09: //TAMPER
+				sendEvent([name: "tamper", value: "detected"])
+				break
+			case 0x01: //ON
+				if (state.soundTable[1]) {
+					sendEvent([name: "alarm", value: "both"])
+				} else {
+					resetActiveSounds()
+					String childDni = "${device.deviceNetworkId}:$state.lastTriggeredSound"
+					def child = childDevices.find { it.deviceNetworkId == childDni }
+					child?.sendEvent([name: "alarm", value: "both"])
+				}
+				break
+			case 0x00: //OFF
+					String childDni = "${device.deviceNetworkId}:$state.lastTriggeredSound"
+					def child = childDevices.find { it.deviceNetworkId == childDni }
+					child?.sendEvent([name: "alarm", value: "off"])
+					sendEvent([name: "alarm", value: "off"])
+					sendEvent([name: "tamper", value: "clear"])
+				break
+		}
+	}
+}
+
+def setOnChild(deviceDni, value) {
+	state.lastTriggeredSound = deviceDni?.split(":")[-1] as Integer
+	setActiveSound(state.lastTriggeredSound)
+	resetActiveSounds()
```

Comment:
Instead of looping, why not just send the "off" message to the previous `lastTriggeredSound` child before you overwrite the value?

I don't think you need all this state that you're keeping around. The index of the last triggered sound should be sufficient. This seems more complicated than it needs to be unless I'm missing something.